rclcpp_components/include/rclcpp_components/component_manager_isolated.hpp

```diff
@@ -67,8 +64,8 @@ class ComponentManagerIsolated : public rclcpp_components::ComponentManager
     exec->add_node(node_wrappers_[node_id].get_node_base_interface());
     executor_wrapper.executor = exec;
     executor_wrapper.thread = std::thread(
-      [exec, cancel_token = executor_wrapper.promise.get_future()]() {
-        exec->spin_until_future_complete(cancel_token);
+      [exec]() {
+        exec->spin();
```

Comment:
this is also bug fix to take advantage of `rclcpp::executors::MultiThreadedExecutor` with `--use_multi_threaded_executor` option? before this fix, it is not multi-threaded to execute the executables.

Reply:
I didn't realize that there was this option. I can have a look in a follow up PR.